src/createAnimatedComponent/JSPropsUpdater.ts

```diff
@@ -0,0 +1,153 @@
+  ) {
+    const viewTag = findNodeHandle(animatedComponent);
+    JSPropsUpdaterPaper._tagToComponentMapping.set(viewTag, animatedComponent);
+    if (JSPropsUpdaterPaper._tagToComponentMapping.size === 1) {
+      const listener = (data: ListenerData) => {
+        const component = JSPropsUpdaterPaper._tagToComponentMapping.get(
+          data.viewTag
+        );
+        component?._updateFromNative(data.props);
+      };
+      this._reanimatedEventEmitter.addListener(
+        'onReanimatedPropsChange',
+        listener
+      );
+    }
+  }
+
+  public removeOnJSPropsChangeListener(
+    animatedComponent: React.Component<unknown, unknown>
+  ) {
+    const viewTag = findNodeHandle(animatedComponent);
+    JSPropsUpdaterPaper._tagToComponentMapping.delete(viewTag);
+    if (JSPropsUpdaterPaper._tagToComponentMapping.size === 0) {
+      this._reanimatedEventEmitter.removeAllListeners(
+        'onReanimatedPropsChange'
+      );
+    }
+  }
+}
+
+class JSPropsUpdaterFabric implements JSPropsUpdater {
+  private static _tagToComponentMapping = new Map();
+  private static isInitialized = false;
+
+  constructor() {
+    if (nativeShouldBeMock()) {
+      return;
+    }
+    if (!JSPropsUpdaterFabric.isInitialized) {
+      const updater = (viewTag: number, props: unknown) => {
+        const component =
+          JSPropsUpdaterFabric._tagToComponentMapping.get(viewTag);
+        component?._updateFromNative(props);
+      };
+      runOnUIImmediately(() => {
+        'worklet';
+        global.updateJSProps = (viewTag: number, props: unknown) => {
+          runOnJS(updater)(viewTag, props);
+        };
+      })();
+      JSPropsUpdaterFabric.isInitialized = true;
+    }
+  }
+
+  public addOnJSPropsChangeListener(
+    animatedComponent: React.Component<unknown, unknown>
+  ) {
+    if (!JSPropsUpdaterFabric.isInitialized) {
+      return;
+    }
+    const viewTag = findNodeHandle(animatedComponent);
+    JSPropsUpdaterFabric._tagToComponentMapping.set(viewTag, animatedComponent);
+  }
+
+  public removeOnJSPropsChangeListener(
+    animatedComponent: React.Component<unknown, unknown>
+  ) {
+    if (!JSPropsUpdaterFabric.isInitialized) {
+      return;
+    }
+    const viewTag = findNodeHandle(animatedComponent);
+    JSPropsUpdaterFabric._tagToComponentMapping.delete(viewTag);
+  }
+}
+
+class JSPropsUpdaterWeb implements JSPropsUpdater {
+  public addOnJSPropsChangeListener(
+    _animatedComponent: React.Component<unknown, unknown>
+  ) {
+    // noop
+  }
+
+  public removeOnJSPropsChangeListener(
+    _animatedComponent: React.Component<unknown, unknown>
+  ) {
+    // noop
+  }
+}
+
+function getJSPropsUpdater():
+  | typeof JSPropsUpdaterWeb
+  | typeof JSPropsUpdaterFabric
+  | typeof JSPropsUpdaterPaper {
+  if (shouldBeUseWeb()) {
+    return JSPropsUpdaterWeb;
+  } else if (global._IS_FABRIC) {
+    return JSPropsUpdaterFabric;
+  } else {
+    return JSPropsUpdaterPaper;
+  }
+}
+export default getJSPropsUpdater();
```

Comment:
Do we even need a function here? Why not just export a value like we do in `NativeMethods.ts`?